src/connector/src/parser/unified/json.rs

```diff
@@ -385,10 +386,16 @@ impl JsonParseOptions {
                     .map_err(|_| create_error())?
                     .into()
             }
-
-            (Some(DataType::Decimal), ValueType::String) => ScalarImpl::Decimal(
-                Decimal::from_str(value.as_str().unwrap()).map_err(|_err| create_error())?,
-            ),
+            (Some(DataType::Decimal), ValueType::String) => {
+                let str = value.as_str().unwrap();
+                // the following values are special string generated by postgres and should be handled separately
+                match str {
+                    "NAN" => ScalarImpl::Decimal(Decimal::NaN),
+                    "POSITIVE_INFINITY" => ScalarImpl::Decimal(Decimal::PositiveInf),
+                    "NEGATIVE_INFINITY" => ScalarImpl::Decimal(Decimal::NegativeInf),
+                    _ => ScalarImpl::Decimal(RustDecimal::from_str(str).map(Decimal::Normalized).map_err(|_err| create_error())?),
```

Comment:
Those all-capital values are caused by debezium. When the upstream is not debezium, we shall still be able to parse `NaN` / `Infinity` / `-Infinity`. Let's keep `Decimal::from_str` rather than switch to `Decimal::Normalized(RustDecimal::from_str)`

Reply:
Updated. Thanks!